contracts/stabilizer/EthPegStabilityModule.sol

```diff
@@ -0,0 +1,71 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity ^0.8.4;
+
+import "./PegStabilityModule.sol";
+
+/// @notice ETH PSM that allows separate pausing of mint and redeem
+/// by the guardian and governor
+contract EthPegStabilityModule is PegStabilityModule {
+
+    /// @notice boolean switch that indicates whether redemptions are paused
+    bool public redeemPaused;
+
+    /// @notice event that is emitted when redemptions are paused
+    event RedemptionsPaused(address account);
+    
+    /// @notice event that is emitted when redemptions are unpaused
+    event RedemptionsUnpaused(address account);
+
+    constructor(
+        OracleParams memory params,
+        uint256 _mintFeeBasisPoints,
+        uint256 _redeemFeeBasisPoints,
+        uint256 _reservesThreshold,
+        uint256 _feiLimitPerSecond,
+        uint256 _mintingBufferCap,
+        IERC20 _underlyingToken,
+        IPCVDeposit _surplusTarget
+    ) PegStabilityModule(
+        params,
+        _mintFeeBasisPoints,
+        _redeemFeeBasisPoints,
+        _reservesThreshold,
+        _feiLimitPerSecond,
+        _mintingBufferCap,
+        _underlyingToken,
+        _surplusTarget
+    ) {}
+
+    /// @notice modifier that allows execution when redemptions are not paused
+    modifier whileRedemptionsNotPaused {
+        require(!redeemPaused, "EthPSM: Redeem paused");
+        _;
+    }
+
+    /// @notice modifier that allows execution when redemptions are paused
+    modifier whileRedemptionsPaused {
+        require(redeemPaused, "EthPSM: Redeem not paused");
+        _;
+    }
+
+    /// @notice set secondary pausable methods to paused
+    function pauseRedeem() public onlyGuardianOrGovernor whileRedemptionsNotPaused {
+        redeemPaused = true;
+        emit RedemptionsPaused(msg.sender);
+    }
+
+    /// @notice set secondary pausable methods to unpaused
+    function unpauseRedeem() public onlyGuardianOrGovernor whileRedemptionsPaused {
+        redeemPaused = false;
+        emit RedemptionsUnpaused(msg.sender);
+    }
```

Comment:
add isGovernorOrGuardianOrAdmin or whatever. Also should that modifier be renamed to only* instead of is* for consistency

Reply:
i can add isGovernorOrGuardianOrAdmin, but I can't modify the base CoreRef class modifier without blowing up other contracts.

Reply:
I could extend coreref to have correct naming on the isGovernorOrGuardianOrAdmin modifier.

Reply:
just do isGovernorOrGuardianOrAdmin whatever